src/test/java/in/projecteka/consentmanager/clients/IdentityServiceClientTest.java

```diff
@@ -56,6 +57,42 @@ public void shouldThrowExceptionIfTokenCallFails() {
                 .verify();
     }
 
+    @Test
+    public void shouldThrowExceptionForExpiredOTP() {
+        when(exchangeFunction.exchange(captor.capture())).thenReturn(
+                Mono.just(ClientResponse
+                        .create(HttpStatus.UNAUTHORIZED)
+                        .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .body("{\n" +
+                                "    \"error\": \"1003\",\n" +
+                                "    \"error_description\": \"Invalid Otp\"\n" +
+                                "}")
+                        .build()));
+        StepVerifier.create(identityServiceClient.getToken(formData))
+                .expectErrorMatches(throwable -> throwable instanceof ClientError &&
+                        ((ClientError) throwable).getHttpStatus().is4xxClientError() &&
+                        ((ClientError) throwable).getError().getError().getCode().equals(ErrorCode.OTP_EXPIRED))
+                .verify();
+    }
+
+    @Test
+    public void shouldThrowExceptionForIncorrectOTP() {
+        when(exchangeFunction.exchange(captor.capture())).thenReturn(
+                Mono.just(ClientResponse
+                        .create(HttpStatus.UNAUTHORIZED)
+                        .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .body("{\n" +
+                                "    \"error\": \"1002\",\n" +
+                                "    \"error_description\": \"Invalid Otp\"\n" +
+                                "}")
```

Comment:
If we are using the body string quite of the times can we have some helper function taking code and description and return the body